Common/Tests/Insights/HyperwalletInsightsTests.swift

```diff
@@ -75,38 +105,62 @@ class HyperwalletInsightsTests: XCTestCase {
         XCTAssertNotNil(HyperwalletInsights.shared, "HyperwalletInsights should be initialized")
         let params = [countryTag: country, currencyTag: currency]
         HyperwalletInsights.shared.trackImpression(pageName: pageName, pageGroup: pageGroup, params: params)
+        sleep(2)
+        XCTAssertNotNil(hyperwalletInsights?.insights, "Insights should be reloaded if nil")
+    }
+
+    func testTrackError_InsightsNotInitialized() {
+        hyperwalletInsights?.insights = nil
+        XCTAssertNotNil(HyperwalletInsights.shared, "HyperwalletInsights should be initialized")
+        let errorInfo = ErrorInfoBuilder(type: "errorInfo_type", message: "errorInfo_message")
+            .fieldName("errorInfo_fieldName")
+            .description("errorInfo_description")
+            .code("errorInfo_code")
+            .build()
+        HyperwalletInsights.shared.trackError(pageName: pageName, pageGroup: pageGroup, errorInfo: errorInfo)
+        sleep(2)
```

Comment:
Why sleep instead of expectation?

Reply:
After the method `trackError` execution, property `insights` is a real instance, not a mock.